### PR TITLE
call FlushEvents when closing client, fix config fetch handling

### PR DESCRIPTION
**Pull request by nsalamad**

### CODE REVIEW

**Comment by nsalamad** [comment]
Current dependencies on/for this PR:
* main
  * **PR #70** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/70" target="_blank">Graphite</a> 
    * **PR #71** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/71" target="_blank">Graphite</a>  👈
      * **PR #78** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/78" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/71?utm_source=stack-comment).

File: DevCycle.SDK.Server.Cloud/Api/DVCCloudClient.cs [hunk 1]
```diff
@@ -182,7 +182,7 @@ public async Task<DVCResponse> TrackAsync(User user, Event userEvent)
             }
         }
 
-        public override void Dispose()
+        public override async ValueTask DisposeAsync()
```
Comment (ajwootto) [inline DevCycle.SDK.Server.Cloud/Api/DVCCloudClient.cs]: seems like this was added in C# 8. Does using this increase our minimum supported language version?

File: DevCycle.SDK.Server.Local/Api/DVCLocalClient.cs [hunk 2]
```diff
@@ -233,6 +233,8 @@ public void FlushEvents()
         
         public override void Dispose()
         {
+            Task flushTask = eventQueue.FlushEvents();
```
Comment (ajwootto) [inline DevCycle.SDK.Server.Local/Api/DVCLocalClient.cs]: does triggering this also prevent new events from being tracked at all? the other sdks have that protection built in

**Review by JamieSinn — APPROVED** [review]

**Comment by JamieSinn** [comment]
@ajwootto - the tests seem to be upset with this change - might need to fix them?

**Pull request merged**